lib/ClangImporter/ImportDecl.cpp

```diff
@@ -7676,10 +7676,12 @@ void ClangImporter::Implementation::importAttributes(
 
   if (auto method = dyn_cast<clang::ObjCMethodDecl>(ClangDecl)) {
     if (method->isDirectMethod() && !AnyUnavailable) {
-      auto attr = AvailableAttr::createPlatformAgnostic(
-          C, "", "", PlatformAgnosticAvailabilityKind::UnavailableInSwift);
-      MappedDecl->getAttrs().add(attr);
-      AnyUnavailable = true;
+      if (!isa<AbstractFunctionDecl>(MappedDecl)) {
+        auto attr = AvailableAttr::createPlatformAgnostic(
+            C, "", "", PlatformAgnosticAvailabilityKind::UnavailableInSwift);
+        MappedDecl->getAttrs().add(attr);
+        AnyUnavailable = true;
+      }
```

Comment:
Hmm.  When do we import an ObjCMethodDecl as something besides an AbstractFunctionDecl?  If this triggers for something like properties, we really should support it rather than making it unavailable.

Reply:
I suspect for properties we should still be mapping the ObjCMethodDecl over to the AccessorDecl on the VarDecl and not the VarDecl itself, but I could be wrong.

Reply:
For properties and subscripts, they don't seem to hit this error and I haven't found anything that does. Should we remove this? I kept in the diagnostic in case there was an edge case I didn't know about.

Reply:
I would suggest changing the `dyn_cast` to a `cast` to make it assert if we ever hit this case.